### PR TITLE
Model validations

### PR DESCRIPTION
**Pull request by istanfield09**
[Relevant Pivotal Tracker story.](https://www.pivotaltracker.com/story/show/102723774)

Is it correct that some of these validations should `allow_blank`, or should I amend the tests in order to get them to pass?

Thanks!

### CODE REVIEW

File: app/models/api_token.rb [hunk 1]
```diff
@@ -9,6 +9,8 @@ class ApiToken < ActiveRecord::Base
 
   # TODO validates :access_token, presence: true
   validates :approval_id, presence: true
+  validates :access_token, presence: true, allow_blank: true, length: { minimum: 0, maximum: 255 }
+  validates :user, presence: true, allow_blank: true
```
Comment (afeld) [inline app/models/api_token.rb]: I think we want to require the association with the user...?

File: app/models/approval.rb [hunk 2]
```diff
@@ -16,6 +16,7 @@ class Approval < ActiveRecord::Base
 
   scope :individual, -> { where(type: 'Approvals::Individual') }
 
+  validates :proposal, presence: true, allow_blank: true
```
Comment (afeld) [inline app/models/approval.rb]: @cmc333333 Will all Approvals be associated with a Proposal?

Also, I think `presence: true, allow_blank: true` is basically canceling itself out :wink:

File: app/models/observation.rb [hunk 3]
```diff
@@ -4,6 +4,9 @@ class Observation < ActiveRecord::Base
   belongs_to :proposal
   belongs_to :user
 
+  validates :proposal, presence: true 
+  validates :user, presence: true, allow_blank: true
```
Comment (afeld) [inline app/models/observation.rb]: Shouldn't `allow_blank`.

File: app/models/approval.rb (hunk 2 above)
Comment (cmc333333) [inline app/models/approval.rb]: At this point in time, yep. Hypothetically, we might want to create some sort of "model" approval chain in the future, but that's a bit away.

Comment (istanfield09) [inline app/models/approval.rb]: Haha thanks. I was a little confused writing some of these, and figured allowing blank would be defeating the purpose. If I write a validation, and it causes the test to fail with the reason being that `:proposal` cant be blank, should I change the test?

Comment (afeld) [inline app/models/approval.rb]: Probably...? Might need to create associated models in the fixtures by default or something.

**Pull request closed without merging**